src/frontend/src/common/analytics/eventTypes.ts

```diff
@@ -259,3 +268,23 @@ export type AnalyticsActiveGroupChange = {
   // The group ID that the user is switching to viewing/acting as.
   new_group_id: number;
 };
+
+/** EVENT_TYPES.SAMPLES_FILTER */
+export type AnalyticsSamplesFilter = {
+  // User is filtering by uploadDate
+  filtering_by_upload_date: boolean;
+  // User is filtering by collectionDate
+  filtering_by_collection_date: boolean;
+  // User is filtering by lineage
+  filtering_by_lineage: boolean;
+  // User is filtering by qcStatus
+  filtering_by_qc_status: boolean;
+  // Upload dates that user is filtering on
+  upload_date_start_end: FilterDate;
```

Comment:
Continuing the discussion about `FilterDate` here, I think the best thing is to break out each of the filter chunks into its pieces, that way the payload stays flatter. So we'd have
```
upload_date_start: string;
upload_date_end: string;
...etc...
```

Reply:
The one thing I'm not sure about is _maybe_ expanding the `EventValue` type to include a plain `Date` type. There's a lot of auto-magic stuff going on in how Segment lands data over in Snowflake. Segment is basically auto-generating migrations as it stuffs data into Snowflake. A new event type or modifying an existing event type means it also creates a corresponding migration so the data can land. But it really abstracts a lot of that away from you as the developer. I _think_ a Date would just land as a string (or maybe the auto-magic recognizes ISO8601 strings?), but Segment is ultimately sending JSON payloads itself to their own API, so my guess is that a Date type would not actually help, and gives us a false sense of how things will work.

A lot of how I structured stuff with the `EventValue` and the payloads was intended to avoid running into problems with the auto-magic. I was trying to de-magic our interaction so we'd be less likely to make a reasonable-but-wrong assumption for how data would ultimately land in the analytics DB.

Reply:
makes sense, i can adjust that